src/qibo/tests/test_quantum_info.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pytest
+
+from qibo.quantum_info import *
+
+
+def test_shannon_entropy_errors():
+    with pytest.raises(ValueError):
+        p = np.asarray([1.0, 0.0])
+        shannon_entropy(p, -2)
+    with pytest.raises(TypeError):
+        p = np.asarray([[1.0], [0.0]])
+        shannon_entropy(p)
+    with pytest.raises(TypeError):
+        p = np.asarray([])
+        shannon_entropy(p)
+    with pytest.raises(ValueError):
+        p = np.asarray([1.0, -1.0])
+        shannon_entropy(p)
+    with pytest.raises(ValueError):
+        p = np.asarray([1.1, 0.0])
+        shannon_entropy(p)
+    with pytest.raises(ValueError):
+        p = np.asarray([0.5, 0.4999999])
+        shannon_entropy(p)
+
+
+@pytest.mark.parametrize("base", [2, 10, np.e, 5])
+def test_shannon_entropy(backend, base):
+    prob_array = np.asarray([1.0, 0.0])
+    result = shannon_entropy(prob_array, base)
+    backend.assert_allclose(result, 0.0)
```

Comment:
```suggestion
    prob_array = backend.cast(np.asarray([1.0, 0.0]))
    result = shannon_entropy(prob_array, base)
    backend.assert_allclose(result, 0.0)
```

Reply:
Added `backend.cast()` to all arrays. Also added `dtype` to them, otherwise `backend.cast()` would default to `complex128` and throw exceptions on TensorFlow.

Even though I do have cupy installed locally, I'm not sure if it was tested with no errors or if it wasn't tested.